_overviews/hello-scala/classes.md

```diff
@@ -0,0 +1,261 @@
+---
+layout: multipage-overview
+title: Scala Classes
+description: This page shows examples of how to create Scala classes, including the basic Scala class constructor.
+partof: hello_scala
+overview-name: Hello, Scala
+discourse: true
+num: 19
+previous-page: try-catch-finally
+next-page: classes-aux-constructors
+---
+
+
+In support of object-oriented programming (OOP), Scala provides a *class* construct. The syntax is much more concise than languages like Java and C#, but it’s also still easy to use and read.
+
+
+
+## Basic class constructor
+
+Here’s a Scala class whose constructor defines two parameters, `firstName` and `lastName`:
+
+```scala
+class Person(var firstName: String, var lastName: String)
```

Comment:
Here’s a bit of my rationale and thought process:

I organized it this way because I came to Scala from an OOP background, and my initial interest was in using Scala as a “better Java.” So I thought basic classes and OOP features might be something other Java developers would want to see early, and it would make the initial transition easier for the typical Java/OOP developer.

So I guess I came to Scala because it looked like a more concise version of Java, kind of a mix of Ruby and Java from what I knew at the time, and then over time — over a period of years, actually — I learned about the FP features, primarily because I kept wondering why the collection methods were written the way they are.

Similarly, I just started working with Dart a few weeks ago, and my initial questions were:

- What’s the syntax for creating a class?
- How do getters/setters work?
- Do they have public/private/protected?
- What do they do about inheritance with traits and/or abstract classes?
- Is there anything like a case class?

I guess that’s how my brain works when I look at new languages. I don’t know if other Java developers think that way, but I assumed that they did.

I also tend to use plain classes when writing OOP code, such as for Swing/JavaFX GUI code, and only use case classes with FP, so I thought it made sense to introduce case classes later in the FP section.

Ideally the book will evolve over time to conclude with some lessons on the FP/OOP fusion ideas presented in *Programming in Scala*, but I haven’t thought about that too much yet.

That’s probably more than you wanted to know :) but I thought I’d share my thought process.

Reply:
Thanks a lot for the detailed explanation :+1: